gvfs-helper.c

```diff
@@ -2105,16 +2107,22 @@ static void do_throttle_spin(struct gh__request_params *params,
 	strbuf_addstr(&region, gh__server_type_label[params->server_type]);
 	trace2_region_enter("gvfs-helper", region.buf, NULL);
 
-	progress = start_progress(progress_msg, duration);
+	if (gh__cmd_opts.show_progress)
+		progress = start_progress(progress_msg, duration);
+
 	while (now < end) {
-		display_progress(progress, (now - begin));
+		if (gh__cmd_opts.show_progress)
```

Comment:
Nit: there is no need to check `if (gh__cmd_opts.show_progress)` here because `display_progress` will no-op when `progress` is `NULL`.

Reply:
good point.  thanks!  always feels odd passing null pointers around and trusting the callee to do the right thing....

Reply:
> always feels odd passing null pointers around and trusting the callee to do the right thing....

I know the feeling!  Truth be told I learned that the checked could be skipped in feedback I received on my midx --[no]-progress changes (where I was also checking before calling `display_progress`).